oci/private/index.sh.tpl

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+set -o pipefail -o errexit -o nounset
+
+readonly YQ="{{yq_path}}"
+
+
+function add_image() {
+    local image_path="$1"
+    local output_path="$2"
+
+    local manifests=$("${YQ}" eval '.manifests[]' "${image_path}/index.json")
+
+    for manifest in "${manifests}"; do
+        local manifest_blob_path=$("${YQ}" '.digest | sub(":"; "/")' <<< ${manifest})
+        local config_blob_path=$("${YQ}" '.config.digest | sub(":"; "/")' "${image_path}/blobs/${manifest_blob_path}")
+
+        local platform=$("${YQ}" --output-format=json '{"os": .os, "architecture": .architecture, "variant": .variant, "os.version": .["os.version"], "os.features": .["os.features"]} | with_entries(select( .value != null ))' "${image_path}/blobs/${config_blob_path}")
+
+        platform="${platform}" \
+        manifest="${manifest}" \
+        "${YQ}" --inplace --output-format=json '.manifests += [env(manifest) + {"platform": env(platform)}]' "${output_path}/manifest_list.json"
+    done
+}
+
+function copy_blob() {
+    local image_path="$1"
+    local output_path="$2"
+    local blob_image_relative_path="$3"
+    local dest_path="${output_path}/${blob_image_relative_path}"
+    mkdir -p "$(dirname "${dest_path}")"
+    cat "${image_path}/${blob_image_relative_path}" > "${dest_path}"
+}
+
+function create_oci_layout() {
+    local path="$1"
+    mkdir -p "${path}"
+
+    echo '{"imageLayoutVersion": "1.0.0"}' > "${path}/oci-layout" 
+    echo '{"schemaVersion": 2, "manifests": []}' > "${path}/index.json"
+    echo '{"schemaVersion": 2, "mediaType": "application/vnd.oci.image.index.v1+json", "manifests": []}' > "${path}/manifest_list.json"
+}
+
+CURRENT_IMAGE=""
+OUTPUT=""
+
+for ARG in "$@"; do
+    case "$ARG" in
+        (--output=*) OUTPUT="${ARG#--output=}"; create_oci_layout "$OUTPUT" ;;
+        (--image=*) CURRENT_IMAGE="${ARG#--image=}"; add_image "$CURRENT_IMAGE" "$OUTPUT" ;;
+        (--blob=*) copy_blob "${CURRENT_IMAGE}" "$OUTPUT" "${ARG#--blob=}" ;;
+        (*) echo "Unknown argument ${ARG}"; exit 1;;
+    esac
+done
+
+
+export checksum=$(shasum -a 256 "${OUTPUT}/manifest_list.json" | cut -f 1 -d " ")
```

Comment:
you could get the sha256 has of the manifest_list.json file with an action and make it more portable since some systems may not have `shasum` installed?

https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/hash/hash.bzl

Reply:
check `cli/brew/brew_artifacts.bzl` in silo for an example of how to use that bit of starlark that ships with Bazel in a rule

Reply:
looks like it depends on python rules. I don't feel good about using that. I'll look for alternatives.

Reply:
especially with python3 python2 mess bazel has. maybe https://github.com/jonelo/jacksum/releases is a more sensible choice here.

Reply:
sounds good. carrying a python dep is bad.

tho I'm also fine with landing this as is since most linux distros will have shasum and we can fix it as a follow-up.

Reply:
I'll do some looking to replace it with a hermetic tool.  https://github.com/jonelo/jacksum/releases looks good. though it requires a java_binary target. How do you feel about having dep on built-in java rules?

Reply:
filed https://github.com/bazel-contrib/rules_oci/issues/41